query/s3/s3_bucket_restrict_public_read_access.sql

```diff
@@ -9,22 +9,48 @@ with data as (
     and (
       grants ->> 'Permission' = 'FULL_CONTROL'
       or grants ->> 'Permission' = 'READ_ACP'
+      or grants ->> 'Permission' = 'READ'
     )
-  )
+  ),read_access_policy as (
+    select
+      distinct name
+    from
+      aws_s3_bucket,
+      jsonb_array_elements(policy_std -> 'Statement') as s,
+      jsonb_array_elements_text(s -> 'Action') as action
+    where
+      s ->> 'Effect' = 'Allow'
+      and (
+        s -> 'Principal' -> 'AWS' = '["*"]'
+        or s ->> 'Principal' = '*'
+      )
+      and (
+        action = '*'
+        or action = '*:*'
+        or action = 's3:*'
+        or action ilike 's3:get%'
+        or action ilike 's3:describe%'
```

Comment:
the IAM permission for List contents is "ListBucket", AFAIK, there are no S3 Describe calls. 

https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html#amazons3-ListBucket